src/hooks/useDisclosure.ts

```diff
@@ -0,0 +1,18 @@
+import { useBoolean } from "usehooks-ts"
+
+/**
+ * Hook that provides a more semantic API for managing the open/close state of a
+ * modal, dropdown, or any other component that can be opened and closed.
+ */
+function useDisclosure(defaultValue = false) {
+  const { value, setTrue, setFalse, toggle } = useBoolean(defaultValue)
+
+  return {
+    isOpen: value,
+    onOpen: setTrue,
+    onClose: setFalse,
+    onToggle: toggle,
+  }
+}
+
+export default useDisclosure
```

Comment:
We use named exports for our other hooks. Would suggest we flip this for consistency

```suggestion
export { useDisclosure }
```